src/js/common/schemaform/FormPage.jsx

```diff
@@ -56,57 +24,29 @@ function focusForm() {
 class FormPage extends React.Component {
   constructor(props) {
     super(props);
-    this.validate = this.validate.bind(this);
     this.onChange = this.onChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
-    this.onError = this.onError.bind(this);
     this.goBack = this.goBack.bind(this);
     this.getEligiblePages = this.getEligiblePages.bind(this);
-    this.getEmptyState = this.getEmptyState.bind(this);
-    this.transformErrors = this.transformErrors.bind(this);
-    this.state = this.getEmptyState(props.route.pageConfig);
   }
 
   componentDidMount() {
-    if (!this.props.reviewPage) {
-      scrollToTop();
-      focusForm();
-    } else if (this.props.reviewPage && this.props.reviewMode) {
-      focusForm();
-    }
-  }
-
-  componentWillReceiveProps(newProps) {
-    if (newProps.route.pageConfig.pageKey !== this.props.route.pageConfig.pageKey) {
-      this.setState(this.getEmptyState(newProps.route.pageConfig), () => {
-        focusForm();
-      });
-    }
+    focusForm();
   }
 
   componentDidUpdate(prevProps) {
     if (prevProps.route.pageConfig.pageKey !== this.props.route.pageConfig.pageKey) {
-      scrollToTop();
+      focusForm();
     }
   }
 
-  onChange({ formData }) {
+  onChange(formData) {
     this.props.setData(this.props.route.pageConfig.pageKey, formData);
   }
 
-  onError() {
-    const formContext = _.set('submitted', true, this.state.formContext);
-    this.setState({ formContext });
-    scrollToFirstError();
-  }
-
   onSubmit() {
-    if (this.props.reviewPage) {
-      this.props.onSubmit();
```

Comment:
Is the `onSubmit` received by props used anymore? If not, it can probably be taken out of the `propTypes` (line 130).